src/Acuminator/Acuminator.Vsix/AcuminatorVSPackage.cs

```diff
@@ -244,25 +249,32 @@ protected override void Dispose(bool disposing)
 
 			if (ThreadHelper.CheckAccess() && _dteSolutionEvents != null)
 			{		
-				_dteSolutionEvents.AfterClosing -= SolutionEvents_AfterClosing;		
+				_dteSolutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
+                _dteSolutionEvents.Opened -= SolutionEvents_Opened;
 			}
 		}
 
-		private async System.Threading.Tasks.Task InitializeSuppressionManagerAsync(IProgress<ServiceProgressData> progress)
+		private System.Threading.Tasks.Task InitializeSuppressionManagerWithProgressAsync(IProgress<ServiceProgressData> progress)
 		{
 			var progressData = new ServiceProgressData(VSIXResource.PackageLoad_WaitMessage, VSIXResource.PackageLoad_InitSuppressionManager,
 													   currentStep: 3, TotalLoadSteps);
 			progress?.Report(progressData);
 
-			var workspace = await this.GetVSWorkspaceAsync();
-			var additionalFiles = workspace.CurrentSolution.Projects
-														   .SelectMany(p => p.AdditionalDocuments)
-														   .Select(d => new SuppressionManagerInitInfo(d.FilePath, generateSuppressionBase: false));
+            return InitializeSuppressionManagerAsync();
+        }
+
+        private async System.Threading.Tasks.Task InitializeSuppressionManagerAsync()
+        {
+            var workspace = await this.GetVSWorkspaceAsync();
+            var additionalFiles = workspace.CurrentSolution.Projects
+                .SelectMany(p => p.AdditionalDocuments)
+                .Select(d => new SuppressionManagerInitInfo(d.FilePath, generateSuppressionBase: false));
+
+            SuppressionManager.Init(new SuppressionFileSystemService(), additionalFiles);
```

Comment:
@metallmonster everything looks fine. However, there is one thing bugging me.
 If I understand correctly you create a new suppression manager during the re-opening of the solution with its own file service and file watcher. 
Suppose we open for example same solution, or another solution but with the same projects.
The old file service will still perform an operation on a suppression file while a new one will try to access it. I'm not sure if this is entirely possible but still I want to discuss it. Maybe we need to lock in the old fileservice to wait until it finishes its job.
@dropsonic what do you think about the possibility?

Reply:
@metallmonster @SENya1990 a user can open two instances of Visual Studio that look at the same project, so I think it should be possible to edit this file simultaneously.

Reply:
done